bliss/simulator/simulated_dataset.py

```diff
@@ -98,19 +98,31 @@ def simulate_image(
             and psf parameters
         """
         images, psfs, psf_params, wcs_batch = self.image_decoder.render_images(
-            tile_catalog, image_ids
+            tile_catalog, image_ids, self.coadd_depth
         )
+        batch_size = images.shape[0]
+        assert images.ndim == 5, f"Expected `images` to be 5D, got {images.ndim}D."
+        # (1) align
+        for b in range(batch_size):
+            first_batch_wcs = wcs_batch[b]
+            for d in range(self.coadd_depth):
+                images[b, d] = torch.from_numpy(
+                    align(
+                        images[b, d].numpy(),
+                        ref_wcs=first_batch_wcs,
+                        ref_band=self.catalog_prior.b_band,
+                    )
+                )
+        images = torch.squeeze(images, dim=1)  # remove coadd depth dimension if 1
+        # (2) coadd via inverse-variance weighting
+        if self.coadd_depth > 1:
+            coadded_images = np.zeros((batch_size, *images.shape[-3:]))  # 4D
+            for b in range(batch_size):
+                coadded_images[b] = self.survey.coadd_images(images[b])
+            images = torch.from_numpy(coadded_images).float()
```

Comment:
It's hard to follow what's going on here. Would you add some comments and/or break this up into smaller functions? It might be better to move "_apply_noise" and this coaddition code to `decoder.py`.

Reply:
Broke it up into smaller functions—leaving `_apply_noise` and coaddition in simulator for now.